src/host/PtySignalInputThread.cpp

```diff
@@ -273,25 +284,14 @@ bool PtySignalInputThread::_GetData(_Out_writes_bytes_(cbBuffer) void* const pBu
                                     const DWORD cbBuffer)
 {
     DWORD dwRead = 0;
-    // If we failed to read because the terminal broke our pipe (usually due
-    //      to dying itself), close gracefully with ERROR_BROKEN_PIPE.
-    // Otherwise throw an exception. ERROR_BROKEN_PIPE is the only case that
-    //       we want to gracefully close in.
     if (FALSE == ReadFile(_hFile.get(), pBuffer, cbBuffer, &dwRead, nullptr))
     {
-        auto lastError = GetLastError();
-        if (lastError == ERROR_BROKEN_PIPE)
-        {
-            _Shutdown();
-        }
-        else
-        {
-            THROW_WIN32(lastError);
-        }
+        return false;
     }
-    else if (dwRead != cbBuffer)
+
+    if (dwRead != cbBuffer)
     {
-        _Shutdown();
+        return false;

```

Comment:
Now that `RundownAndExit` is missing, we aren't `[[noreturn]]` anymore and need to return something. Almost didn't catch this due to the disabled linter checks.